include/zephyr/net/coap_client.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/net/coap.h>
+
+/**
+ * @typedef coap_client_response_cb_t
+ * @brief Type of the callback being called when there is response to the CoAP request.
+ *
+ * @param result_code Result code of the response. Negative if there was a failure in send.
+ *                    \ref enum coap_response_code for positive.
+ * @param offset Offset starting from 0 for the payload.
+ * @param payload Buffer containing the payload from the response.
+ * @param len Size of the payload.
+ * @param user_data User provided context.
+ * @param last_block Indicates the last block of the response.
+ */
+typedef void (*coap_client_response_cb_t)(uint8_t result_code,
+					  size_t offset, size_t len, const uint8_t *payload,
+					  bool last_block, void *user_data);
+
+/**
+ * @brief Representation of a CoAP client request.
+ */
+struct coap_client_request {
+	enum coap_method method; /** Method of the request */
+	bool confirmable; /** CoAP Confirmable/Non-confirmable message */
+	const char *path; /** Path of the requested resource */
+	enum coap_content_format fmt; /** Content format to be used */
+	uint8_t *payload; /** User allocated buffer for send request */
+	size_t len; /** Length of the payload */
+	coap_client_response_cb_t cb; /** Callback when response received */
+	void *user_data; /** User provided context */
+};
```

Comment:
```suggestion
struct coap_client_request {
	enum coap_method method;      /**< Method of the request */
	bool confirmable;	      /**< CoAP Confirmable/Non-confirmable message */
	const char *path;	      /**< Path of the requested resource */
	enum coap_content_format fmt; /**< Content format to be used */
	uint8_t *payload;	      /**< User allocated buffer for send request */
	size_t len;		      /**< Length of the payload */
	coap_client_response_cb_t cb; /**< Callback when response received */
	void *user_data;	      /**< User provided context */
};
```

Please align the comment blocks. Also you need to use `/**<` when the Doxygen block refers to the previous member.